tests/python/pants_test/engine/legacy/test_owners_integration.py

```diff
@@ -0,0 +1,25 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants_test.pants_run_integration_test import PantsRunIntegrationTest
+
+
+class ListOwnersIntegrationTest(PantsRunIntegrationTest):
+  def test_owner_of_success(self):
+    pants_run = self.do_command('--owner-of=testprojects/tests/python/pants/dummies/test_pass.py',
+                                'list',
+                                success=True)
+    self.assertEquals(
+      pants_run.stdout_data.strip(),
+      'testprojects/tests/python/pants/dummies:passing_target'
+    )
+
+  def test_owner_list_not_owned(self):
+    pants_run = self.do_command('--owner-of=testprojects/tests/python/pants/dummies/test_nonexistent.py',
+                                'list',
+                                success=True)
+    self.assertIn('WARNING: No targets were matched in', pants_run.stderr_data)
```

Comment:
A third case to check here might be when two targets own the same files.

Reply:
Mmm, so we should assert that both targets are listed correctly? If you know of a file like that in `testprojects` that'd be helpful. Otherwise I'll look for an already existing one and if there isn't I might as well create another target in some build_file I guess, but I rather not because I don't want to break other things 👀

Reply:
Hm. I think we should list both targets in that case, but I'm not 100% sure. I don't know whether there's a case like that in testprojects.

I think there was discussion a while ago about disallowing targets from owning the same sources, but as far as I know, we don't do that right now. You could also file an issue for it and defer.